### PR TITLE
Plugin Modernization

### PR DESCRIPTION
**Pull request by gounthar**
I have followed most of the [Improve a Plugin Tutorial](https://www.jenkins.io/doc/developer/tutorial-improve/).

### Testing done

```bash
mvn clean install
mvn clean verify
```

with maven `3.9.4` and JDK `11` and `17`.

### Submitter checklist
- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did
- ~~Link to relevant issues in GitHub or Jira~~
- ~~Link to relevant pull requests, esp. upstream and downstream changes~~
- [X] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by MarkEWaite** [comment]
I launched the ci.jenkins.io job with the new Jenkinsfile.

**Comment by gounthar** [comment]
Thanks a lot, Mark, we'll see if this build is better than the previous one. 👍

**Comment by gounthar** [comment]
All the checks passed. 🚀 
I think the previous build error was due to an incompatibility of some sort between our current BOM dependencies and JDK8 (super wild guess, I have no data to prove it).

File: pom.xml [hunk 1]
```diff
@@ -116,10 +116,12 @@
         </pluginRepository>
     </pluginRepositories>
 
-    <properties>
-        <jenkins.version>2.277.1</jenkins.version>
+    <properties>
+        <revision>2.9.4</revision>
+        <changelist>-SNAPSHOT</changelist>
+        <gitHubRepo>jenkinsci/startup-trigger-plugin</gitHubRepo>
+        <jenkins.version>2.401.3</jenkins.version>
```
Comment (MarkEWaite) [inline pom.xml:123]: I think that the `jenkins.version` and the `bom` version should match.
```suggestion
        <jenkins.version>2.387.3</jenkins.version>

```

**Review by MarkEWaite — CHANGES_REQUESTED** [review]
Minimum Jenkins version should match with the bom version

**Comment by gounthar** [comment]
Thank you, Mark.
It appears that the issue arose after initially following the tutorial and subsequently entrusting Dependabot with my fork without a thorough review.

In brief, I made an error, and I am committed to rectifying it promptly.

**Comment by MarkEWaite** [comment]
> entrusting Dependabot with my fork without a thorough review

That is an interesting challenge.  I am surprised that dependabot offered an update of the `jenkins.version` value.  I've never seen that.

**Comment by gounthar** [comment]
Yes, that doesn't make much sense to me either...
I will check the list of commits and PRs on my fork and see where I failed.
Spoiler alert: I guess in the end I will find I'm the only one who's guilty, and the bot is innocent.

**Comment by MarkEWaite** [comment]
No problem.  https://github.com/jenkinsci/startup-trigger-plugin/pull/8/commits/fec66f9f2615801646814550168863b5be0bd9af shows that when updating to make Jenkins 2.401.3 the minimum version, the bom line remained at `bom-2.387.x`.  No problem if the plugin chooses to require 2.401.3 or 2.387.3 as its minimum Jenkins version, but I think that it is healthy to keep the bom version matched with the Jenkins minimum version.

**Comment by gounthar** [comment]
Thank you, Mark.

Now, I understand. It's because I followed the tutorial "[Update minimum required Jenkins version](https://www.jenkins.io/doc/developer/tutorial-improve/update-base-jenkins-version/#update-minimum-required-jenkins-version)" without thinking. The tutorial instructs you to update the version like this:

```xml
   <properties>
     <jenkins.version>2.401.3</jenkins.version>
   </properties>
```

and then to update the bill of materials accordingly... Using a diff such as:

 ```xml
 <dependencyManagement>
    <dependencies>
      <dependency>
        <groupId>io.jenkins.tools.bom</groupId>
-        <artifactId>bom-2.319.x</artifactId>
+        <artifactId>bom-2.387.x</artifactId>
-        <version>1466.v85a_616ea_b_87c</version>
+        <version>2179.v0884e842b_859</version>
        <type>pom</type>
        <scope>import</scope>
      </dependency>
    </dependencies>
  </dependencyManagement>
```

So, I noticed the version mentioned was 2.387, and I might have thought to myself, "I'm good; I have the same thing as the example is showing." However, I should have considered consistency and used the same version base. 😊

Is this part of the documentation something we should aim to keep updated using updatecli, or is it sufficient as it stands now? I mean, most Jenkins plugin developers will likely connect the dots automatically, which I, unfortunately, didn't manage to do. 🤷

**Pull request merged**